Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,494 @@
+                "using SearchResolution to divide into equally spaced line segments. Successive iterations will then " +
+                "subdivide further the nearest segment, producing more accurate results. In most cases, the default value (2) is sufficient.")]
+            public int m_SearchIteration;
+        }
+
+        /// <summary>Controls how automatic dollying occurs</summary>
+        [Tooltip("Controls how automatic dollying occurs.  A Follow target is necessary to use this feature.")]
+        public AutoDolly m_AutoDolly = new AutoDolly 
+        { 
+            m_Enabled = false, 
+            m_PositionOffset = 0,
+            m_SearchIteration = 4, 
+            m_SearchResolution = 2,
+        };
+
+        /// <summary>True if component is enabled and has a spline</summary>
+        public override bool IsValid => enabled && m_Spline != null;
+
+        /// <summary>Get the Cinemachine Pipeline stage that this component implements.
+        /// Always returns the Body stage</summary>
+        public override CinemachineCore.Stage Stage => CinemachineCore.Stage.Body;
+
+        /// <summary>
+        /// Report maximum damping time needed for this component.
+        /// </summary>
+        /// <returns>Highest damping setting in this component</returns>
+        public override float GetMaxDampTime() => !m_DampingEnabled ? 0 :
+            Mathf.Max(Mathf.Max(m_Damping.x, Mathf.Max(m_Damping.y, m_Damping.z)), m_AngularDamping);
+
+        /// <summary>
+        /// Adds spline to SplineDolly.
+        /// </summary>
+        /// <param name="spline">The spline to add. If null, then any spline on SplineDolly will be removed.</param>
+        public void AddSplineContainer(SplineContainer spline)
+        {
+            m_Spline = spline;
+            UpdateSplineData();
+        }
+
+        /// <summary>
+        /// Subscribe to onSplineChanged if you'd like to react to changes to the Spline attached to this vcam.
+        /// This action is invoked by the Spline's changed event when a spline property is modified. Available in editor only.
+        /// </summary>
+        public event Action onSplineChanged;
+
+        /// <summary>Positions the virtual camera according to the transposer rules.</summary>
+        /// <param name="curState">The current camera state</param>
+        /// <param name="deltaTime">Used for damping.  If less that 0, no damping is done.</param>
+        public override void MutateCameraState(ref CameraState curState, float deltaTime)
+        {
+            if (!IsValid)
+            {
+                return;
+            }
+
+            var spline = m_Spline.Spline;
+            if (spline == null || spline.Count == 0)
+            {
+                return;
+            }
+
+#if false
+            m_CameraPosition = spline.SanitizeInterpolationValue(m_CameraPosition, m_PositionUnits);
+#else
```

Comment:
This is the API that spline team might add. Might be different, I just did a PR to their repo.